lit_gpt/datasets/tinyllama.py

```diff
@@ -0,0 +1,99 @@
+# Copyright Lightning AI. Licensed under the Apache License 2.0, see LICENSE file.
+
+from pathlib import Path
+from typing import Union, Optional
+
+from torch.utils.data import DataLoader
+
+from lit_gpt import Tokenizer
+from lit_gpt.datasets import LitDataModule
+
+
+class TinyLlama(LitDataModule):
+    """The TinyLlama data module is composed of a mix of SlimPajama and Starcoder data.
+
+    Provides training and validation streaming dataloaders that return batches of tokens.
+
+    Args:
+        data_path: The path to the data directory, containing two folders 'slimpajama' and 'starcoder'
+            which are the output of the preprocessing step done in advance. See the `tutorial/pretrain_tinyllama.md`
+            for instructions. The path can also be a remote path (e.g., s3://).
+        seed: The seed to use for shuffling the training datasets.
+        num_workers: The number of workers to use for the dataloaders.
+    """
+
+    def __init__(
+        self,
+        data_path: Union[str, Path] = Path("data/"),
```

Comment:
Let's only accept paths, from CLI this doesn't matter.

```suggestion
        data_path: Path = Path("data/"),
```